vue-app/src/api/core.ts

```diff
@@ -1,7 +1,22 @@
 import { ethers } from 'ethers'
+import WalletConnectProvider from '@walletconnect/web3-provider'
 
 import { FundingRoundFactory } from './abi'
 
+export const providers = {
+  metamaskProvider: new ethers.providers.StaticJsonRpcProvider(
+    process.env.VUE_APP_ETHEREUM_API_URL
+  ),
+  walletconnectProvider: new WalletConnectProvider({
+    infuraId: '4299898b85054e0386e7136ab3170d11',
+    rpc: {
+      // 1: "http://localhost:18545",
+      3: 'https://rinkeby.infura.io/v3/',
+    },
+    // chainId: 1
+  }),
+}
+
 export const provider = new ethers.providers.StaticJsonRpcProvider(
```

Comment:
Did you have a chance to check what is going on with this exported provider? because it is being used in a lot of places.
As before we were only using one provider it didn't matter, but now we have a dynamic provider, which could be MM or any other. Probably we should think about how we can share the "selected" provider by the user, right?

Reply:
Yes, that's a great app example to follow. Would be nice to implement it as they did, creating a plugin to provide and inject the provider. https://github.com/balancer-labs/frontend-v2/blob/develop/src/main.ts#L52